src/cli/vector_enrich_doc.rs

```diff
@@ -0,0 +1,352 @@
+use crate::state::State;
+use log::debug;
+use std::time::SystemTime;
+
+use crate::cli::llm_client::LLMClients;
+use crate::cli::util::read_openai_api_key;
+use crate::CtrlcFuture;
+use crate::OpenAIClient;
+use nu_protocol::Example;
+use nu_protocol::Record;
+use std::convert::TryFrom;
+use std::str;
+use std::sync::{Arc, Mutex};
+use tokio::runtime::Runtime;
+use tokio::select;
+
+use nu_engine::CallExt;
+
+use nu_protocol::ast::Call;
+use nu_protocol::engine::Command;
+use nu_protocol::engine::{EngineState, Stack};
+use nu_protocol::{
+    Category, IntoPipelineData, PipelineData, ShellError, Signature, SyntaxShape, Value,
+};
+
+#[derive(Clone)]
+pub struct VectorEnrichDoc {
+    state: Arc<Mutex<State>>,
+}
+
+impl VectorEnrichDoc {
+    pub fn new(state: Arc<Mutex<State>>) -> Self {
+        Self { state }
+    }
+}
+
+impl Command for VectorEnrichDoc {
+    fn name(&self) -> &str {
+        "vector enrich-doc"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("vector enrich-doc")
+            .required(
+                "field",
+                SyntaxShape::String,
+                "the field from which the vector is generated",
+            )
+            .named(
+                "dimension",
+                SyntaxShape::Int,
+                "dimension of the resulting embeddings (default 128)",
+                None,
+            )
+            .named(
+                "maxTokens",
+                SyntaxShape::Int,
+                "the token per minute limit with 'text-embedding-3-small' for your API key",
+                None,
+            )
+            .named(
+                "id-column",
+                SyntaxShape::String,
+                "the name of the id column if used with an input stream",
+                None,
+            )
+            .category(Category::Custom("couchbase".to_string()))
+    }
+
+    fn usage(&self) -> &str {
+        "Enriches given JSON with embeddings of selected field"
+    }
+
+    fn run(
+        &self,
+        engine_state: &EngineState,
+        stack: &mut Stack,
+        call: &Call,
+        input: PipelineData,
+    ) -> Result<PipelineData, ShellError> {
+        vector_enrich_doc(self.state.clone(), engine_state, stack, call, input)
+    }
+
+    fn examples(&self) -> Vec<Example> {
+        vec![
+            Example {
+                description: "Open local json doc and enrich the field named 'description'",
+                example: "open ./local.json | vector enrich-doc",
```

Comment:
Why doesn't think require a field param?

Reply:
It does, I forgot it